skel/rust-api/src/main.rs

```diff
@@ -0,0 +1,74 @@
+use std::net::{Ipv4Addr, SocketAddr};
+
+use axum::{routing, Router, Server};
+use hyper::Error;
+use rust_api::file;
+use utoipa::{
+    openapi::security::{ApiKey, ApiKeyValue, SecurityScheme},
+    Modify, OpenApi,
+};
+use utoipa_swagger_ui::SwaggerUi;
+
+use tower_http::trace::{self, TraceLayer};
+use tracing::{ info, Level };
+//use tracing_subscriber::{layer::SubscriberExt, util::SubscriberInitExt};
+
+#[tokio::main]
+async fn main() -> Result<(), Error> {
+    tracing_subscriber::fmt()
+        .with_target(false)
+        .compact()
+        //.json()
+        .init();
+
+    // // TRACE
+    // tracing_subscriber::registry()
+    //     .with(tracing_subscriber::fmt::layer())
+    //     .init();
+
+    #[derive(OpenApi)]
+    #[openapi(
+        paths(
+            file::search,
+        ),
+        components(
+            schemas(file::File, file::FileError)
+        ),
+        modifiers(&SecurityAddon),
+        tags(
+            (name = "File", description = "File manager API")
+        )
+    )]
+    struct ApiDoc;
+
+    #[derive(Clone)]
+    struct AppState {}
+    struct SecurityAddon;
+
+    impl Modify for SecurityAddon {
```

Comment:
Would it be better if these were outside of the main function?

Reply:
Thought exactly the same right after merging it, along with putting a few more comments to guide new user's edits :-S

Reply:
https://github.com/umccr/orcabus/commit/483226ff149feddfd155e82b3a9823ef60df1752